src/test/java/com/connexta/store/controllers/StoreControllerIngestTests.java

```diff
@@ -0,0 +1,229 @@
+        .andExpect(status().isBadRequest());
+
+    verifyNoInteractions(mockStoreService);
+  }
+
+  @Test
+  public void testMissingLastModifiedHeader() throws Exception {
+    mockMvc
+        .perform(
+            multipart("/ingest")
+                .file(FILE)
+                .file(METACARD)
+                .header("Accept-Version", ACCEPT_VERSION)
+                .param("correlationId", CORRELATION_ID))
+        .andExpect(status().isBadRequest());
+
+    verifyNoInteractions(mockStoreService);
+  }
+
+  @ParameterizedTest
+  @ValueSource(
+      strings = {
+        "2011-12-03", // ISO_LOCAL_DATE
+        "2011-12-03+01:00", // ISO_OFFSET_DATE
+        "10:15:30", // ISO_LOCAL_TIME
+        "10:15:30+01:00", // ISO_OFFSET_TIME
+        "2011-12-03T10:15:30", // ISO_LOCAL_DATE_TIME
+        "2012-337", // ISO_ORDINAL_DATE
+        "2012-W48-6", // ISO_WEEK_DATE
+        "20111203", // BASIC_ISO_DATE
+        "Sun, 11 Jun 2017 14:32:28 GMT", // RFC_1123_DATE_TIME
+        "2017-06-11T14:32:28.120+0000", // ISO_DATE_TIME missing offset :
+        "             ",
+        ""
+      })
+  public void testInvalidLastModifiedHeader(String badDate) throws Exception {
+    mockMvc
+        .perform(
+            multipart("/ingest")
+                .file(FILE)
+                .file(METACARD)
+                .header("Accept-Version", ACCEPT_VERSION)
+                .header(LAST_MODIFIED, badDate)
+                .param("correlationId", CORRELATION_ID))
+        .andExpect(status().isBadRequest());
+
+    verifyNoInteractions(mockStoreService);
+  }
+
+  /**
+   * TODO Test RuntimeException and Throwable thrown by {@link StoreService#ingest(Long, String,
+   * InputStream, String, Long, InputStream)}
+   */
+  @ParameterizedTest(name = "{0} is the response status code when IngestService#ingest throws {1}")
+  @MethodSource("exceptionThrownByIngestServiceAndExpectedResponseStatus")
+  public void testIngestServiceExceptions(
+      final Throwable throwable, final HttpStatus expectedResponseStatus) throws Exception {
+    doThrow(throwable).when(mockStoreService).ingest(any(), any(), any(), any(), any(), any());
+
+    mockMvc
+        .perform(
+            multipart("/ingest")
+                .file(FILE)
+                .file(METACARD)
+                .param("correlationId", CORRELATION_ID)
+                .header(LAST_MODIFIED, LAST_MODIFIED_DATE)
+                .header("Accept-Version", ACCEPT_VERSION))
+        .andExpect(status().is(expectedResponseStatus.value()));
+  }
+
+  private static Stream<Arguments> exceptionThrownByIngestServiceAndExpectedResponseStatus() {
+    return Stream.of(
+        Arguments.of(new StoreException("Test", new Throwable()), HttpStatus.INTERNAL_SERVER_ERROR),
+        Arguments.of(new TransformException(new Throwable()), HttpStatus.INTERNAL_SERVER_ERROR),
+        Arguments.of(
+            new StoreException("Test", new Throwable()), HttpStatus.INTERNAL_SERVER_ERROR));
```

Comment:
This last argument is the same as the first

Reply:
🗑